src/main/java/uk/gov/pay/directdebit/mandate/params/MandateSearchParams.java

```diff
@@ -101,18 +107,47 @@ public MandateSearchParams withToDate(ZonedDateTime toDate) {
         return this;
     }
 
-    public MandateSearchParams withPage(int page) {
-        this.page = page;
+    public MandateSearchParams withPage(Integer page) {
+        if (page != null && page < 1) {
+            throw new BadRequestException("Query param 'page' should be a non zero positive integer.");
+        }
+        this.page = Optional.ofNullable(page).orElse(0);
         return this;
     }
 
-    public MandateSearchParams withDisplaySize(int displaySize) {
-        this.displaySize = displaySize;
+    public MandateSearchParams withDisplaySize(Integer displaySize) {
+        int size = Optional.ofNullable(displaySize).orElse(500);
+        if (!Range.closed(1, 500).contains(size)) {
+            throw new BadRequestException("Query param 'display_size' should be between 1 and 500.");
+        }
+        this.displaySize = size;
         return this;
     }
 
     public MandateSearchParams withGatewayAccountId(String gatewayAccountId) {
         this.gatewayAccountExternalId = gatewayAccountId;
         return this;
     }
+
+    public MandateSearchParams withFromDate(String fromDate) {
+        if (isNotBlank(fromDate)) {
+            try {
+                withFromDate(ZonedDateTime.parse(fromDate));
+            } catch (DateTimeParseException e) {
+                throw new BadRequestException(format("Input from_date (%s) is wrong format.", fromDate));
+            }
+        }
+        return this;
+    }
+
+    public MandateSearchParams withToDate(String toDate) {
+        if (isNotBlank(toDate)) {
+            try {
+                withToDate(ZonedDateTime.parse(toDate));
+            } catch (DateTimeParseException e) {
+                throw new BadRequestException(format("Input to_date (%s) is wrong format.", toDate));
+            }
+        }
+        return this;
+    }
```

Comment:
These two methods could probably share common code in a private method

Reply:
true